### DISCUSSION TITLE
exposeRefs - a way for access of composition API's ref values

### DISCUSSION THREAD

**Discussion by cyfung1031** [quasarframework/quasar#10777] [Ideas / Proposals]
In the composition API designs, there are many ref values are stored inside the ui components. 

Example:
```
setup(){
  const hiddenState = ref(false)
  return {
    setHiddenState (val) {
      hiddenState.value=val
    }
  }
}
```
You can just set the hidden state via some exposed public function but you cannot know the state value.

Those states are very important if we want to customize the behavior of the UI.

For example, 
* QDrawer's pulling position
* QDrawer's actual showing (a bit different from the v-model value)
* QDrawer's actual width (it is different between mini-state and normal)

Why don't just give the option to programmers to see this value?

I propose to use a boolean `exposeRefs` defined in config.js so it can allow programmers to access these state properties.

`app.use(Quasar, { config: { exposeRefs: true } })`

The actual implementation can be found in the PR #10761. It is fully backwards compatible and does not affecting any existing usage.

**Comment by metalsadman** [comment]
Good idea, though what needs to be exposed should be discussed first and be agreed upon, then through specific feat request imho.
